clients/unshaded/src/main/java/tachyon/client/block/LocalBlockInStream.java

```diff
@@ -0,0 +1,149 @@
+  private final ByteBuffer mData;
+
+  private boolean mClosed;
+
+  public LocalBlockInStream(long blockId) throws IOException {
+    mBlockId = blockId;
+    mClosed = false;
+    mContext = BSContext.INSTANCE;
+    mWorkerClient =
+        mContext.acquireWorkerClient(NetworkAddressUtils.getLocalHostName(ClientContext.getConf()));
+    String blockPath = mWorkerClient.lockBlock(blockId);
+
+    if (null == blockPath) {
+      // TODO: Handle this error case better
+      mContext.releaseWorkerClient(mWorkerClient);
+      throw new IOException("Block is not available on local machine");
+    }
+
+    // Map the data to the blockData byte buffer
+    Closer closer = Closer.create();
+    try {
+      RandomAccessFile localFile = closer.register(new RandomAccessFile(blockPath, "r"));
+      long fileLength = localFile.length();
+      FileChannel localFileChannel = closer.register(localFile.getChannel());
+      mData = localFileChannel.map(FileChannel.MapMode.READ_ONLY, 0, fileLength);
+    } finally {
+      closer.close();
+    }
+  }
+
+  @Override
+  public void close() throws IOException {
+    if (mClosed) {
+      return;
+    }
+    mWorkerClient.unlockBlock(mBlockId);
+    mContext.releaseWorkerClient(mWorkerClient);
+    // TODO: Evaluate if this is necessary
+    BufferUtils.cleanDirectBuffer(mData);
+    mClosed = true;
+  }
+
+  @Override
+  public int read() throws IOException {
+    failIfClosed();
+    if (mData.remaining() == 0) {
+      close();
+      return -1;
+    }
+    return BufferUtils.byteToInt(mData.get());
+  }
+
+  @Override
+  public int read(byte[] b) throws IOException {
+    failIfClosed();
+    return read(b, 0, b.length);
+  }
+
+  @Override
+  public int read(byte[] b, int off, int len) throws IOException {
+    failIfClosed();
+    Preconditions.checkArgument(b != null, "Buffer is null");
+    Preconditions.checkArgument(off >= 0 && len >= 0 && len + off <= b.length, String
+        .format("Buffer length (%d), offset(%d), len(%d)", b.length, off, len));
+    if (len == 0) {
+      return 0;
+    }
+
+    int ret = Math.min(len, mData.remaining());
+    if (ret == 0) {
+      close();
+      return -1;
+    }
+    mData.get(b, off, ret);
+    return ret;
+  }
+
+  @Override
+  public long remaining() {
+    return mData.remaining();
+  }
+
+  public void seek(long pos) throws IOException {
+    failIfClosed();
+    Preconditions.checkArgument(pos >= 0, "Seek position is negative: " + pos);
+    Preconditions.checkArgument(pos <= mData.limit(), "Seek position is past buffer limit: " + pos
+        + ", Buffer Size = " + mData.limit());
+    mData.position((int) pos);
+  }
+
+  @Override
+  public long skip(long n) throws IOException {
+    failIfClosed();
+    if (n <= 0) {
+      return 0;
+    }
+
+    int ret = mData.remaining();
+    if (ret > n) {
+      ret = (int) n;
+    }
+    mData.position(mData.position() + ret);
+    return ret;
+  }
+
+  private void failIfClosed() throws IOException {
```

Comment:
NIT: how about naming this as `checkNotClosed` to keep consistent with other part of the code?